### PR TITLE
MSEND - send message to multiple users

### PR DESCRIPTION
**Pull request by larsbrinkhoff**
SYS2; TS MSEND
SYSENG;MSEND 140

### CODE REVIEW

**Comment by larsbrinkhoff** [comment]
The reconstructed TAA; INP 100 (version picked arbitrarily) assembles identically to the corresponding parts in the SYS2; TS MSEND binary.

There's some question as to how the symbols should be defined.  The binary has a separate symbol block called READER which comes from INP.  But some of the symbols defined in the INP file are located in the MSEND block.  I have reconstructed this by putting most of INP inside .BEGIN READER ... .END, but defining some symbols with .U" which is the syntax to go "up" one symbol block level.

The reconstructed TAA; UUOS 100 had a TITLE.  This overwrites the TITLE in MSEND, resulting in a symbol block called UUOS and no MSEND block.  I changed it to a SUBTTL which restores the MSEND block.

As noted above, the assembled MSEND binary will not run.  I will add a fix and bump the version.

File: src/taa/inp.100 [hunk 1]
```diff
@@ -0,0 +1,747 @@
...
+RCMD0:	SETOM	INREAD
+RCMD1:	.IOT	TTYI,A
+	SKIPE	RQUOTE
+	 JRST	[SETZM	RQUOTE
+		 SKIPE	KEEPQT
+		  JRST	RCMDL
+		 SKIPN	TOERS
+		  JRST	RCMDL
+		 PUSH	P,A
+		 MOVEI	A,21
```
Comment (larsbrinkhoff) [inline src/taa/inp.100]: Maybe `^Q`?  I haven't checked what the code does, but RQUOTE seems to indiate something with quoting.

File: src/taa/inp.100 [hunk 2]
```diff
@@ -0,0 +1,747 @@
...
+	RCMDL
+	RCMDXE
+
+RCMDXE:	.IOT	TTYO,A
+RCMDX1:	MOVEM	A,LSTBRK
+	MOVEI	A,0
+	IDPB	A,B
+	ADD	B,[070000,,0]
+	TLNE	B,400000
+	 ADD	B,[347777777777]
```
Comment (larsbrinkhoff) [inline src/taa/inp.100]: Should use the DBP macro.

File: src/taa/inp.100 [hunk 3]
```diff
@@ -0,0 +1,747 @@
...
+		 DSKI
+		 SETZM	A]
+	 .LOSE	%LSSYS
+	.CALL	[SETZ
+		 SIXBIT	/OPEN/
+		 MOVSI	.UAI
+		 DSKI
+		 [SIXBIT /ERR/]
+		 MOVEI	3
+		 SETZ	A]
```
Comment (larsbrinkhoff) [inline src/taa/inp.100]: FIY: FN1 = 3 is a magic number which makes the ERR device accept a code in FN2.

File: src/taa/inp.100 [hunk 4]
```diff
@@ -0,0 +1,747 @@
...
+	POPJ	P,
+
+CNTTAB:	ANDCMI	O,7
+	ADDI	O,10
+	MOVEI	C,0
+	POPJ	P,
+
+FIXIML:	111111175641
+	131111111111
+	112011120000
```
Comment (larsbrinkhoff) [inline src/taa/inp.100]: These octal digits index into the FIXIM2 table.

File: src/taa/inp.100 [hunk 5]
```diff
@@ -0,0 +1,747 @@
...
+.U"BUFNCH==3
+.U"BUFLEN==4
+.U"BUFVLN==5
+.U"BUFCCT==6
+.U"BUFVCT==7
+.U"BUFHLP==10
+
+RCMDNX:	MOVEM	B,BUFNCH(CBUF)
+	SKIPGE	C
+	 MOVEI	3,0
```
Comment (atsampson) [inline src/taa/inp.100]: MOVEI C,0

File: src/taa/inp.100 (hunk 1 above)
Comment (atsampson) [reply src/taa/inp.100]: It feeds it into CHRTYP, so yes, ^Q.

File: src/taa/inp.100 [hunk 6]
```diff
@@ -0,0 +1,747 @@
...
+	SETOM	ENDFIL
+	MOVEI	F,D
+PDLOOP:	LDB	O,E
+	CAIE	O,^C
+	 CAIN	O,0
+	  JRST	PDFLS
+	JRST	FINSTR
+PDFLS:	MOVEI	O,0
+	DPB	O,E
+	ADD	E,[070000,,0]
```
Comment (atsampson) [inline src/taa/inp.100]: DBP E

File: src/taa/inp.100 [hunk 7]
```diff
@@ -0,0 +1,747 @@
...
+RUB:	PUSHJ	P,RUBBER
+	 JRST	RCMDNX
+	JRST	RCMD1
+
+RUBBER:	SOJL	C,CPOPJ
+	LDB	A,B
+	MOVEI	D,0
+	DPB	D,B
+	XCT	XCTRUB
+	ADD	B,[070000,,0]
```
Comment (atsampson) [inline src/taa/inp.100]: DBP B

File: src/taa/inp.100 [hunk 8]
```diff
@@ -0,0 +1,747 @@
...
+	PUSH	P,A
+	LDB	A,B
+	CAIE	A,^M
+	 JRST	[POP P,A ? JRST RUBOUT]
+	SUB	P,[1,,1]
+	MOVEI	D,0
+	DPB	D,B
+	SUBI	C,1
+	XCT	XCTRUB
+	ADD	B,[070000,,0]
```
Comment (atsampson) [inline src/taa/inp.100]: DBP B

File: src/taa/inp.100 [hunk 9]
```diff
@@ -0,0 +1,747 @@
...
+	POPJ	P,
+
+LNFLUS:	PUSHJ	P,RUBBER
+	 JRST	RCMDNX
+LNFLUL:	LDB	A,B
+	CAIN	A,^J
+	 JRST	LNFLUD
+	MOVEI	O,0
+	DPB	O,B
+	ADD	B,[070000,,0]
```
Comment (atsampson) [inline src/taa/inp.100]: DBP B

File: src/taa/inp.100 [hunk 10]
```diff
@@ -0,0 +1,747 @@
...
+	TLNE	B,400000
+	 ADD	B,[347777777777]
+	SOJLE	C,LNLEAV
+	JRST	LNFLUL
+
+LNFLUD:	PUSH	P,B
+	ADD	B,[070000,,0]
```
Comment (atsampson) [inline src/taa/inp.100]: DBP B

**Comment by larsbrinkhoff** [comment]
@atsampson the "DPB O," is deposit byte so I think the accumulator should stay.

Where DBP is intended, yes there should be no accumulator.

**Comment by larsbrinkhoff** [comment]
Oh sorry, I see you commented on the ADD 070000 line.  Yes, right.

**Comment by larsbrinkhoff** [comment]
Updated with the suggested edits.

**Comment by larsbrinkhoff** [comment]
See #1918

**Pull request merged**